### PR TITLE
Make mpConfig duplicate converters deterministic

### PR DESCRIPTION
**Pull request by Joseph-Cass**
* Updated implementation
* Added tests for mpConfig < 2.0
* Added test for mpConfig > 1.4

### CODE REVIEW

**Comment by Joseph-Cass** [comment]
#build

File: dev/com.ibm.ws.microprofile.config.1.1_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/test/Ant.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*******************************************************************************
+ * Copyright (c) 2016 IBM Corporation and others.
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.config.1.1_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/test/Ant.java:2]: date?

File: dev/com.ibm.ws.microprofile.config.1.1_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/test/AntConverter.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+ * Copyright (c) 2016 IBM Corporation and others.
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.config.1.1_fat/test-applications/cdiConfig.jar/src/com/ibm/ws/microprofile/appConfig/cdi/test/AntConverter.java:2]: date

**Comment by Joseph-Cass** [comment]
#build

**Comment by Joseph-Cass** [comment]
Build failed due to something unrelated. Related tests passed.

**Review by tevans78 — APPROVED** [review]

**Comment by Joseph-Cass** [comment]
#run-libby-bot

**Pull request merged**